src/test/txpackage_tests.cpp

```diff
@@ -441,20 +441,6 @@ BOOST_AUTO_TEST_CASE(package_submission_tests)
         BOOST_CHECK_EQUAL(result_quit_early.m_state.GetResult(), PackageValidationResult::PCKG_TX);
     }
 
-    // Child with missing parent.
```

Comment:
Might still be an interesting case to keep around, shoving it to the end after parent is already accepted:

```
diff --git a/src/test/txpackage_tests.cpp b/src/test/txpackage_tests.cpp
index 09e719ff4c..be0ebac834 100644
--- a/src/test/txpackage_tests.cpp
+++ b/src/test/txpackage_tests.cpp
@@ -476,10 +476,30 @@ BOOST_AUTO_TEST_CASE(package_submission_tests)
             BOOST_CHECK(it_child_deduped->second.m_result_type == MempoolAcceptResult::ResultType::MEMPOOL_ENTRY);
         }
 
         BOOST_CHECK_EQUAL(m_node.mempool->size(), expected_pool_size);
     }
+
+    // In-mempool parent and child with missing parent.
+    mtx_child.vin.emplace_back(COutPoint(package_unrelated[0]->GetHash(), 0));
+    CTransactionRef tx_child_missing_parent = MakeTransactionRef(mtx_child);
+    Package package_missing_parent;
+    package_missing_parent.push_back(tx_parent);
+    package_missing_parent.push_back(tx_child_missing_parent);
+    {
+        const auto result_missing_parent = ProcessNewPackage(m_node.chainman->ActiveChainstate(), *m_node.mempool,
+                                                             package_missing_parent, /*test_accept=*/false, /*client_maxfeerate=*/{});
+        auto it_parent = result_missing_parent.m_tx_results.find(tx_parent->GetWitnessHash());
+        auto it_child = result_missing_parent.m_tx_results.find(tx_child_missing_parent->GetWitnessHash());
+        BOOST_CHECK(result_missing_parent.m_state.IsInvalid());
+        BOOST_CHECK_EQUAL(result_missing_parent.m_state.GetResult(), PackageValidationResult::PCKG_TX);
+        BOOST_CHECK_EQUAL(result_missing_parent.m_state.GetRejectReason(), "transaction failed");
+        BOOST_CHECK(it_parent->second.m_state.IsValid());
+        BOOST_CHECK_EQUAL(it_child->second.m_state.GetResult(), TxValidationResult::TX_MISSING_INPUTS);
+        BOOST_CHECK_EQUAL(it_child->second.m_state.GetRejectReason(), "bad-txns-inputs-missingorspent");
+        BOOST_CHECK_EQUAL(m_node.mempool->size(), expected_pool_size);
+    }
 }
 
 // Tests for packages containing a single transaction
 BOOST_AUTO_TEST_CASE(package_single_tx)
 {
```